x-pack/legacy/plugins/index_lifecycle_management/public/store/selectors/lifecycle.js

```diff
@@ -123,9 +130,9 @@ export const validatePhase = (type, phase, errors) => {
       } else if (
```

Comment:
I was looking through the code again, and I'm wondering if this block of code is necessary. I don't think it will ever reach this point, as we're already checking for negative numbers on [L121](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/index_lifecycle_management/public/store/selectors/lifecycle.js#L121).

```
} else if (phase[numberedAttribute] < 0) {
  phaseErrors[numberedAttribute] = [positiveNumberRequiredMessage];
}
```

Can you confirm?

Reply:
It will reach this point. Initially I thought the negative numbers on L121 would have caught it too, but it didn't.

Reply:
Can you share how you were testing it to reach this?

Reply:
Sure, no problem. Initially I set <=0

```
} else if (phase[numberedAttribute] <= 0) {
  phaseErrors[numberedAttribute] = [positiveNumberRequiredMessage];
}
```

This will generate an error when I enter 0 through the Index Lifecycle Policy cold phase page. It was still not permitting 0. When I set < 0

```
} else if (phase[numberedAttribute] < 0) {
  phaseErrors[numberedAttribute] = [positiveNumberRequiredMessage];
}
```
I will not get an error when I enter 0 on the ILP page on the cold phase. Then when I run the jest test I will get a `received` error of 0 as 0 is now a valid value, but I will get an`expected` error of 1. 

```
expect(received).toBe(expected) // Object.is equality
    Expected: 1
    Received: 0
      74 | const expectedErrorMessages = (rendered, expectedErrorMessages) => {
      75 |   const errorMessages = rendered.find('.euiFormErrorText');
    > 76 |   expect(errorMessages.length).toBe(expectedErrorMessages.length);
         |                                ^
      77 |   expectedErrorMessages.forEach(expectedErrorMessage => {
      78 |     let foundErrorMessage;
      79 |     for (let i = 0; i < errorMessages.length; i++) {
```


Therefore, I edited the jest test case to allow no error to be returned by setting as the value 0 is permitted now

`expectedErrorMessages(rendered, [])`

Reply:
Sorry, I wasn't very clear in my comment. I was referring to your statement:

> Initially I thought the negative numbers on L121 would have caught it too, but it didn't.

I was wondering what steps you were taking in the UI to reach this point. I am not able to reproduce.

Reply:
Sorry, my misunderstanding. Fixed the issue now.